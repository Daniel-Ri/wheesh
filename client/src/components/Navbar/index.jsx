import PropTypes from 'prop-types';
import { useState } from 'react';
import { FormattedMessage } from 'react-intl';
import { useDispatch } from 'react-redux';
import { useLocation, useNavigate } from 'react-router-dom';
import { useMediaQuery } from 'react-responsive';
import Avatar from '@mui/material/Avatar';
import Menu from '@mui/material/Menu';
import MenuItem from '@mui/material/MenuItem';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import ExpandLessIcon from '@mui/icons-material/ExpandLess';
import HomeIcon from '@mui/icons-material/Home';
import ReceiptLongIcon from '@mui/icons-material/ReceiptLong';
import SentimentSatisfiedAltIcon from '@mui/icons-material/SentimentSatisfiedAlt';
import DoorSlidingIcon from '@mui/icons-material/DoorSliding';

import { setLocale } from '@containers/App/actions';

import classes from './style.module.scss';

// eslint-disable-next-line no-unused-vars
const Navbar = ({ title, locale, theme }) => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { pathname } = useLocation();

  const isSMMaxSize = useMediaQuery({ query: '(max-width: 576px)' });

  const [menuPosition, setMenuPosition] = useState(null);
  const open = Boolean(menuPosition);

  const handleClick = (event) => {
    setMenuPosition(event.currentTarget);
  };

  const handleClose = () => {
    setMenuPosition(null);
  };

  const onSelectLang = (lang) => {
    if (lang !== locale) {
      dispatch(setLocale(lang));
    }
    handleClose();
  };

  const goHome = () => {
    navigate('/');
  };

  const goMyTicketsPage = () => {
    navigate('/my-tickets');
  };

  const goGatePage = () => {
    navigate('/gate');
  };

  const goMePage = () => {
    navigate('/me');
  };

  return (
    <div className={classes.headerWrapper} data-testid="navbar">
      <div className={classes.contentWrapper}>
        <div className={classes.logoImage} onClick={goHome}>
          <img src="/train.svg" alt="logo" className={classes.logo} />
          <img src="/wheesh.png" alt="title" className={classes.title} />
        </div>
        <div className={classes.toolbar}>
          <div
            data-testid="HomeLink"
            className={`${classes.link} ${pathname === '/' && classes.linkSelected}`}
            onClick={goHome}
          >
            <HomeIcon className={classes.icon} />
            {!isSMMaxSize && (
              <div className={classes.namePage}>
                <FormattedMessage id="app_home" />
              </div>
            )}
          </div>
          <div
            data-testid="MyTicketsLink"
            className={`${classes.link} ${pathname === '/my-tickets' && classes.linkSelected}`}
            onClick={goMyTicketsPage}
          >
            <ReceiptLongIcon className={classes.icon} />
            {!isSMMaxSize && (
              <div className={classes.namePage}>
                <FormattedMessage id="app_my_tickets" />
              </div>
            )}
          </div>
          <div
            data-testid="GateLink"
            className={`${classes.link} ${pathname === '/gate' && classes.linkSelected}`}
            onClick={goGatePage}
          >
            <DoorSlidingIcon className={classes.icon} />
            {!isSMMaxSize && (
              <div className={classes.namePage}>
                <FormattedMessage id="app_gate" />
              </div>
            )}
          </div>
          <div
            data-testid="MeLink"
            className={`${classes.link} ${pathname === '/me' && classes.linkSelected}`}
            onClick={goMePage}
          >
            <SentimentSatisfiedAltIcon className={classes.icon} />
            {!isSMMaxSize && (
              <div className={classes.namePage}>
                <FormattedMessage id="app_me" />
              </div>
            )}
          </div>
          <div data-testid="ToggleLang" className={classes.toggle} onClick={handleClick}>
            <Avatar className={classes.avatar} src={locale === 'id' ? '/id.png' : '/en.png'} />
            <div className={classes.lang}>{locale}</div>
            {isSMMaxSize ? <ExpandLessIcon /> : <ExpandMoreIcon />}
          </div>
        </div>
        <Menu
          data-testid="MenuLang"
          open={open}
          anchorEl={menuPosition}
          onClose={handleClose}
          transformOrigin={
            isSMMaxSize ? { horizontal: 'right', vertical: 'bottom' } : { horizontal: 'left', vertical: 'top' }
          }
          anchorOrigin={
            isSMMaxSize ? { horizontal: 'right', vertical: 'top' } : { horizontal: 'left', vertical: 'bottom' }
          }
        >
          <MenuItem data-testid="IDItem" onClick={() => onSelectLang('id')} selected={locale === 'id'}>
            <div className={classes.menu}>
              <Avatar className={classes.menuAvatar} src="/id.png" />
              <div className={classes.menuLang}>
                <FormattedMessage id="app_lang_id" />
              </div>
            </div>
          </MenuItem>
          <MenuItem data-testid="ENItem" onClick={() => onSelectLang('en')} selected={locale === 'en'}>
            <div className={classes.menu}>
              <Avatar className={classes.menuAvatar} src="/en.png" />
              <div className={classes.menuLang}>
                <FormattedMessage id="app_lang_en" />
              </div>
            </div>
          </MenuItem>
        </Menu>
      </div>
    </div>
  );
};

Navbar.propTypes = {
  title: PropTypes.string,
  locale: PropTypes.string.isRequired,
  theme: PropTypes.string,
};

export default Navbar;
