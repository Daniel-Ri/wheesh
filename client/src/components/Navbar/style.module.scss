@import '@styles/base';

.headerWrapper {
  background-color: var(--color-bg) !important;
  box-shadow: 0 0.25rem 0.75rem rgba(25, 49, 83, 0.2) !important;
  width: 10rem;
  height: 100vh;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;

  .contentWrapper {
    @include directionY;

    color: var(--color-text);
    padding: 1rem;
    align-items: start;
    justify-content: space-between;
    gap: 1.5rem;
    font-size: 1rem;

    .logoImage {
      @include directionX;
      cursor: pointer;
      align-items: center;

      .logo {
        width: 1.5rem;
        height: 1.5rem;
        margin-right: 0.5rem;
      }
      .title {
        height: 1.25rem;
      }
    }

    .toolbar {
      @include directionY;
      gap: 1.5rem;

      .link {
        @include directionX;
        align-items: center;
        gap: 0.5rem;
        cursor: pointer;

        .namePage {
          font-size: 100%;
        }
      }

      .linkSelected {
        color: $color-picton-blue;
      }

      .theme {
        @include directionX;
        align-items: center;
        margin-right: 0.5rem;
        cursor: pointer;
      }
      .toggle {
        @include directionX;
        align-items: center;
        cursor: pointer;

        .avatar {
          margin-left: 0.1rem;
          width: 1.2rem;
          height: 1.2rem;
          box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
          margin-right: 0.5rem;
        }

        .lang {
          font-family: Poppins, Arial, Helvetica, sans-serif;
          font-weight: 700;
          font-size: 1rem;
          text-transform: uppercase;
        }

        .arrow {
          font-size: 1rem;
        }
      }
    }
  }

  @media screen and (max-width: $screen-sm-limit) {
    top: auto;
    bottom: 0;
    width: 100%;
    height: fit-content;

    .contentWrapper {
      @include directionX;
      align-items: center;
      justify-content: space-around;
      padding: 1rem 0.5rem;

      .logoImage {
        width: 20%;

        .logo {
          margin-right: 0.1rem;
        }

        .title {
          height: 0.75rem;
        }
      }

      .toolbar {
        width: 80%;
        @include directionX;
        justify-content: space-around;
        gap: 0.1rem;

        .link {
          @include directionY;
          gap: 0.1rem;

          .namePage {
            font-size: 70%;
          }
        }

        .theme {
          margin-right: 0;
        }
      }
    }
  }
}

.menu {
  @include directionX;
  align-items: center;

  .menuAvatar {
    width: 1rem;
    height: 1rem;
    box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
    margin-right: 0.5rem;
  }

  .menuLang {
    font-family: Poppins, Arial, Helvetica, sans-serif;
    font-size: 0.8rem;
  }
}
