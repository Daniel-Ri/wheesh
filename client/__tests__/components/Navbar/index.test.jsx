import { render, fireEvent } from '@utils/testHelper';
import Navbar from '@components/Navbar';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  useNavigate: () => mockNavigate,
  useLocation: () => '/',
}));

let wrapper;
beforeEach(() => {
  wrapper = render(<Navbar title="Title" locale="id" theme="light" />);
});

describe('Component Navbar', () => {
  test('Correct render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('navbar')).toBeInTheDocument();
  });

  test('Should call navigate when HomeLink clicked', () => {
    const { getByTestId } = wrapper;
    const homeNav = getByTestId('HomeLink');
    fireEvent.click(homeNav);
    expect(mockNavigate).toHaveBeenCalledWith('/');
  });

  test('Should call navigate when MyTicketsLink clicked', () => {
    const { getByTestId } = wrapper;
    const myTicketsNav = getByTestId('MyTicketsLink');
    fireEvent.click(myTicketsNav);
    expect(mockNavigate).toHaveBeenCalledWith('/my-tickets');
  });

  test('Should call navigate when MeLink clicked', () => {
    const { getByTestId } = wrapper;
    const meNav = getByTestId('MeLink');
    fireEvent.click(meNav);
    expect(mockNavigate).toHaveBeenCalledWith('/me');
  });

  test('Should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
