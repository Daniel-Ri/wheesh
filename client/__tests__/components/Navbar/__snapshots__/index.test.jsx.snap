// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Navbar Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="headerWrapper"
        data-testid="navbar"
      >
        <div
          class="contentWrapper"
        >
          <div
            class="logoImage"
          >
            <img
              alt="logo"
              class="logo"
              src="/train.svg"
            />
            <img
              alt="title"
              class="title"
              src="/wheesh.png"
            />
          </div>
          <div
            class="toolbar"
          >
            <div
              class="link false"
              data-testid="HomeLink"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <div
                class="namePage"
              >
                Beranda
              </div>
            </div>
            <div
              class="link false"
              data-testid="MyTicketsLink"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                data-testid="ReceiptLongIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.5 3.5 18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2v14H3v3c0 1.66 1.34 3 3 3h12c1.66 0 3-1.34 3-3V2l-1.5 1.5zM19 19c0 .55-.45 1-1 1s-1-.45-1-1v-3H8V5h11v14z"
                />
                <path
                  d="M9 7h6v2H9zm7 0h2v2h-2zm-7 3h6v2H9zm7 0h2v2h-2z"
                />
              </svg>
              <div
                class="namePage"
              >
                Tiket Saya
              </div>
            </div>
            <div
              class="link false"
              data-testid="MeLink"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                data-testid="SentimentSatisfiedAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5"
                />
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-2.5c2.33 0 4.32-1.45 5.12-3.5h-1.67c-.69 1.19-1.97 2-3.45 2s-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5z"
                />
              </svg>
              <div
                class="namePage"
              >
                Saya
              </div>
            </div>
            <div
              class="toggle"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="/id.png"
                />
              </div>
              <div
                class="lang"
              >
                id
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="headerWrapper"
      data-testid="navbar"
    >
      <div
        class="contentWrapper"
      >
        <div
          class="logoImage"
        >
          <img
            alt="logo"
            class="logo"
            src="/train.svg"
          />
          <img
            alt="title"
            class="title"
            src="/wheesh.png"
          />
        </div>
        <div
          class="toolbar"
        >
          <div
            class="link false"
            data-testid="HomeLink"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            <div
              class="namePage"
            >
              Beranda
            </div>
          </div>
          <div
            class="link false"
            data-testid="MyTicketsLink"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ReceiptLongIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5 3.5 18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2v14H3v3c0 1.66 1.34 3 3 3h12c1.66 0 3-1.34 3-3V2l-1.5 1.5zM19 19c0 .55-.45 1-1 1s-1-.45-1-1v-3H8V5h11v14z"
              />
              <path
                d="M9 7h6v2H9zm7 0h2v2h-2zm-7 3h6v2H9zm7 0h2v2h-2z"
              />
            </svg>
            <div
              class="namePage"
            >
              Tiket Saya
            </div>
          </div>
          <div
            class="link false"
            data-testid="MeLink"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="SentimentSatisfiedAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="15.5"
                cy="9.5"
                r="1.5"
              />
              <circle
                cx="8.5"
                cy="9.5"
                r="1.5"
              />
              <circle
                cx="15.5"
                cy="9.5"
                r="1.5"
              />
              <circle
                cx="8.5"
                cy="9.5"
                r="1.5"
              />
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-2.5c2.33 0 4.32-1.45 5.12-3.5h-1.67c-.69 1.19-1.97 2-3.45 2s-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5z"
              />
            </svg>
            <div
              class="namePage"
            >
              Saya
            </div>
          </div>
          <div
            class="toggle"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/id.png"
              />
            </div>
            <div
              class="lang"
            >
              id
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Navbar Component Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="headerWrapper"
        data-testid="navbar"
      >
        <div
          class="contentWrapper"
        >
          <div
            class="logoImage"
          >
            <img
              alt="logo"
              class="logo"
              src="/train.svg"
            />
            <img
              alt="title"
              class="title"
              src="/wheesh.png"
            />
          </div>
          <div
            class="toolbar"
          >
            <div
              class="link false"
              data-testid="HomeLink"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                data-testid="HomeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
              <div
                class="namePage"
              >
                Beranda
              </div>
            </div>
            <div
              class="link false"
              data-testid="MyTicketsLink"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                data-testid="ReceiptLongIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.5 3.5 18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2v14H3v3c0 1.66 1.34 3 3 3h12c1.66 0 3-1.34 3-3V2l-1.5 1.5zM19 19c0 .55-.45 1-1 1s-1-.45-1-1v-3H8V5h11v14z"
                />
                <path
                  d="M9 7h6v2H9zm7 0h2v2h-2zm-7 3h6v2H9zm7 0h2v2h-2z"
                />
              </svg>
              <div
                class="namePage"
              >
                Tiket Saya
              </div>
            </div>
            <div
              class="link false"
              data-testid="MeLink"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                data-testid="SentimentSatisfiedAltIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="15.5"
                  cy="9.5"
                  r="1.5"
                />
                <circle
                  cx="8.5"
                  cy="9.5"
                  r="1.5"
                />
                <path
                  d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-2.5c2.33 0 4.32-1.45 5.12-3.5h-1.67c-.69 1.19-1.97 2-3.45 2s-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5z"
                />
              </svg>
              <div
                class="namePage"
              >
                Saya
              </div>
            </div>
            <div
              class="toggle"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
              >
                <img
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="/id.png"
                />
              </div>
              <div
                class="lang"
              >
                id
              </div>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="headerWrapper"
      data-testid="navbar"
    >
      <div
        class="contentWrapper"
      >
        <div
          class="logoImage"
        >
          <img
            alt="logo"
            class="logo"
            src="/train.svg"
          />
          <img
            alt="title"
            class="title"
            src="/wheesh.png"
          />
        </div>
        <div
          class="toolbar"
        >
          <div
            class="link false"
            data-testid="HomeLink"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="HomeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
            <div
              class="namePage"
            >
              Beranda
            </div>
          </div>
          <div
            class="link false"
            data-testid="MyTicketsLink"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="ReceiptLongIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.5 3.5 18 2l-1.5 1.5L15 2l-1.5 1.5L12 2l-1.5 1.5L9 2 7.5 3.5 6 2v14H3v3c0 1.66 1.34 3 3 3h12c1.66 0 3-1.34 3-3V2l-1.5 1.5zM19 19c0 .55-.45 1-1 1s-1-.45-1-1v-3H8V5h11v14z"
              />
              <path
                d="M9 7h6v2H9zm7 0h2v2h-2zm-7 3h6v2H9zm7 0h2v2h-2z"
              />
            </svg>
            <div
              class="namePage"
            >
              Tiket Saya
            </div>
          </div>
          <div
            class="link false"
            data-testid="MeLink"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
              data-testid="SentimentSatisfiedAltIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="15.5"
                cy="9.5"
                r="1.5"
              />
              <circle
                cx="8.5"
                cy="9.5"
                r="1.5"
              />
              <circle
                cx="15.5"
                cy="9.5"
                r="1.5"
              />
              <circle
                cx="8.5"
                cy="9.5"
                r="1.5"
              />
              <path
                d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm0-2.5c2.33 0 4.32-1.45 5.12-3.5h-1.67c-.69 1.19-1.97 2-3.45 2s-2.75-.81-3.45-2H6.88c.8 2.05 2.79 3.5 5.12 3.5z"
              />
            </svg>
            <div
              class="namePage"
            >
              Saya
            </div>
          </div>
          <div
            class="toggle"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular avatar css-1wlk0hk-MuiAvatar-root"
            >
              <img
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/id.png"
              />
            </div>
            <div
              class="lang"
            >
              id
            </div>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
